### ISSUE TITLE
Cookie not in this host's domain

### ISSUE THREAD

**Issue by Krytical13** [Huachao/vscode-restclient#470]
- VSCode Version: 1.41.1 (user setup)
- OS Version: Windows 10 1809 17763.914
- REST Client Version: 0.23.0

Steps to Reproduce:

0. Note that I am a seasoned sysadmin, but pretty new to REST/JSON and API's.

1.  Attempt to use Cloudflare API within REST Client like the following sanitized example:
```
get https://api.cloudflare.com/client/v4/zones/*zoneIDhere*
content-type: application/json

{
    "Authorization": "Bearer:  *tokenHere*",
    "X-Auth-Email": "*emailHere*"
}
```

2. See resulting error

> Cookie not in this host's domain. Cookie:dash.cloudflare.com Request:api.cloudflare.com

Question:
Do I understand correctly, 
Set-Cookie could be a requirement for any Rest Client in this situation?

**Comment by Huachao** [comment]
@Krytical13 the root cause is that the received `Set-Cookie` header contains a cookie whose domain is `dash.cloudflare.com` which mismatches the request's target domain `api.cloudflare.com`. I will update my code to eliminate such error.

**Comment by Huachao** [comment]
@Krytical13 I have published the latest version 0.23.1 and you can verify it.
